### ISSUE TITLE
Accessing attributes of a Never/NoReturn type does not result in an error

### ISSUE THREAD

**Issue by Wizzerinus** [microsoft/pyright#9619]
**Describe the bug**
Mypy and Pyright give significantly different outputs on the following file. Mypy agrees it is an error, and Pyright does not.

Mypy:
```
e.py:5: error: "Never" has no attribute "bad_attr"  [attr-defined]
e.py:5: note: Revealed type is "Any"
```

Pyright:
```
/home/wizzerinus/tests/e.py
  /home/wizzerinus/tests/e.py:5:17 - information: Type of "x.bad_attr" is "NoReturn"
0 errors, 0 warnings, 1 information
```

**Code or Screenshots**
```python
from typing import NoReturn


def something(x: NoReturn):
    reveal_type(x.bad_attr)
```

**VS Code extension or command-line**
Pyright commandline, version 1.1.390.

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

A value of type `Never` can never be populated by a real value, so this function is not callable. Normally, a `Never` type originates from type narrowing, and in those cases, it's preferable to not generate errors for attempts to access attributes or call these values. Doing so will result in false positive errors for code that is determined through static analysis to be unreachable.
